framework/src/media/FileInputDataSource.cpp

```diff
@@ -67,20 +69,55 @@ bool FileInputDataSource::open()
 			return false;
 		}
 
+		containerType = utils::getContainerTypeFromPath(mDataPath);
```

Comment:
In my opinion, we'd better add container types to audio type.
Below parsing logic including stream_parsing also can be moved to header_parsing or newly make something in mediautils.
Also we can add this in switch-case in registerCodec.
Is there any other specific reason?

Reply:
Fixed, thanks